### DISCUSSION TITLE
graphql/envelop/plugins/use-error-handler

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/envelop#1751] [Docs Discussions]
# graphql/envelop/plugins/use-error-handler

GraphQL Envelop Documentation

https://the-guild.dev/graphql/envelop/plugins/use-error-handler

**Comment by paul-vd** [comment]
The `maskErrorFn` seems to be `maskError` and the return type of `MaskError` is Error, but the `err` argument type is unknown causing type missmatches

**Comment by paul-vd** [comment]
The `isGraphQLError` helper has an incorrect typing, currently it would type the error as `Error` instead of `GraphQLError`

```ts
export declare function isGraphQLError(error: unknown): error is Error & {
    originalError?: Error;
};
```

So if you would want to create another error check to see if it has extensions errors, this would cause the typing to fail because  `extensions` are not defined on the `Error` type, for example

```ts
function isGraphQLErrorExtensions(error: unknown): error is GraphQLError & {
    extensions: NonNullable<GraphQLError['extensions']>;
  } {
    return (
      isGraphQLError(error) &&
      error.extensions !== null &&
      error.extensions !== undefined &&
      typeof error?.extensions === 'object'
    );
  }
```

![image](https://user-images.githubusercontent.com/39598117/230143127-71ab5615-7d41-4f1d-b1b7-91a476ff2a40.png)

**Reply by paul-vd** [reply]
This also causes the `path` on the error object to not exist

**Reply by garfieldnate** [reply]
The typing has been changed to `isGraphQLError(val: unknown): val is GraphQLError`.

**Reply by garfieldnate** [reply]
No, I see now, `GraphQLError` is the same as `error is Error & {originalError?: Error;};`, and it's causing type issues for me while accessing `extensions`, as well.

**Comment by garfieldnate** [comment]
This really needs a lot more details. The type for `errors` is:

```ts
errors: {
    errors: readonly Error[] | readonly SerializableGraphQLErrorLike[];
    context: Readonly<DefaultContext>;
    phase: "parse" | "validate" | "context" | "execution";
}
```

* What do all of these fields indicate? What's the difference between `Error` and `SerializableGraphQLErrorLike`?
* What would be the default error formatting if I didn't register a plugin?
* I want to print out the entire incoming GQL query and the field that failed to resolve. Is that possible?

**Reply by garfieldnate** [reply]
If you have a custom context object configured in your `createYoga` and you want to access that context in your `useErrorHandler`, you'll want this in the first line:

```ts
    useErrorHandler((errors) => {
      const context = errors.context.contextValue as YOUR_CUSTOM_CONTEXT_TYPE;
      ...
    }
```
